tests/tests.cpp

```diff
@@ -53,7 +53,7 @@ TEST(exemple, exemple_test) {
 
 TEST(ctor, bad_library) {
     try {
-        dylib lib("./null.so");
+        dylib lib(dylib::working_directory, "null");
```

Comment:
A string whose text is null? Confusing. Better say "no_such_library".

And you could also add a testcase for actually passing nullptr if you like.